src/hivpy/sexual_behaviour.py

```diff
@@ -29,12 +29,18 @@ class SexualBehaviourModule:
     def select_matrix(self, matrix_list):
         return matrix_list[np.random.choice(matrix_list.shape[0])]
 
+    def norm_probs(self, prob_dict: dict):
+        for key, data in prob_dict.items():
+            data /= sum(data)
```

Comment:
Possibly paranoid here but I'm slightly concerned that `norm_probs` updates in place. This works with numpy arrays (or Python lists) because iterating over them works "by reference". If we switch to something else, though (or if numpy changes the implementation, which I think has happened in the past), this might break. Is it better to return a normalised version instead?
```suggestion
    def norm_probs(self, prob_dict: dict):
        return {
            key: data / sum(data)
            for key, data in prob_dict.items()
        }
```
But not convinced it is.

Reply:
This could also be renamed `_norm_probs` if we want to stress that it should be thought of as private.

Reply:
This is true; I had a think about this too and wasn't sure which was the best way to go; I was mostly concerned about a) doing an unnecessary deep copy and b) having the version of the array in `sexual_behaviour_data` differing from the one that's in the `sexual_behaviour` module (un-normalised and normalised respectively). The `data` version shouldn't really be accessed from that point on though, but it does still exist and could easily be referenced accidentally. That said, I think that this will change anyway once we're loading things from yaml files, as then it could just be normalised at load time and we won't have two copies floating around.